app/views/shared/_component_certificate_doc_links.html.erb

```diff
@@ -1,17 +1,23 @@
 <p>
-  <%= t 'user_journey.before_you_start.find_out_more' %>
+  <%= t 'user_journey.find_out_more' %>
   <% if certificate.component_type == COMPONENT_TYPE::MSA %>
     <% if certificate.encryption? %>
-      <%= link_to t('user_journey.before_you_start.rotate_key_and_certificate', component: COMPONENT_TYPE::MSA_SHORT, certificate: CERTIFICATE_USAGE::ENCRYPTION), t('user_journey.urls.msa_encryption') %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::MSA_SHORT, certificate: CERTIFICATE_USAGE::ENCRYPTION), t('user_journey.urls.msa_encryption') %>
     <% else %>
-      <%= link_to t('user_journey.before_you_start.rotate_key_and_certificate', component: COMPONENT_TYPE::MSA_SHORT, certificate: CERTIFICATE_USAGE::SIGNING), t('user_journey.urls.msa_signing') %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::MSA_SHORT, certificate: CERTIFICATE_USAGE::SIGNING), t('user_journey.urls.msa_signing') %>
     <% end %>
-  <% else %>
+  <% elsif certificate.component.vsp  %>
     <% if certificate.encryption? %>
-      <%= link_to t('user_journey.before_you_start.rotate_key_and_certificate', component: COMPONENT_TYPE::VSP_SHORT, certificate: CERTIFICATE_USAGE::ENCRYPTION), t('user_journey.urls.vsp_encryption') %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::VSP_SHORT, certificate: CERTIFICATE_USAGE::ENCRYPTION), t('user_journey.urls.vsp_encryption') %>
     <% else %>
-      <%= link_to t('user_journey.before_you_start.rotate_key_and_certificate', component: COMPONENT_TYPE::VSP_SHORT, certificate: CERTIFICATE_USAGE::SIGNING), t('user_journey.urls.vsp_signing') %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::VSP_SHORT, certificate: CERTIFICATE_USAGE::SIGNING), t('user_journey.urls.vsp_signing') %>
+    <% end %>
+  <% else  %>
+    <% if certificate.encryption? %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::SP_SHORT, certificate: CERTIFICATE_USAGE::ENCRYPTION), t('user_journey.urls.sp_encryption') %>
+    <% else %>
+      <%= link_to t('user_journey.rotate_key_and_certificate', component: COMPONENT_TYPE::SP_SHORT, certificate: CERTIFICATE_USAGE::SIGNING), t('user_journey.urls.sp_signing') %>
```

Comment:
Would be cool to decrease the repetition a bit here. You could put a couple of helper methods in the controller: `helper_method :linkotron` and then do something like:

```
def linkotron(certificate)
  {
     component: display_component(certificate.component),
     certificate: certificate.usage
  }
end
```

```
def url_thingy(certificate)
  "#{certificate.component.underscore}_#{certificate.usage}"
end
```

Then can we just have one link?

```
<%= link_to t('user_journey.rotate_key_and_certificate', t(user_journey.urls.#{url_thingy(certificate)}"), linkotron(certificate) %>
```

The localisation en would have to change to be `user_journey.urls.sp_component_signing`.

I suppose you could do all the above in the view without methods as well, though less logic in the views is best. Haven't tried this, just off the top of my head so if you think it won't work feel free to ignore

Reply:
Got a lovely one liner for you, see what you think
`<%= t('user_journey.find_out_more', link: link_to(t('user_journey.rotate_key_and_certificate', component: display_component(certificate.component), certificate: certificate.usage), t("user_journey.urls.#{display_component(certificate.component)}.#{certificate.usage}"))).html_safe %>`